pytest_plugins/external_logging.py

```diff
@@ -0,0 +1,27 @@
+from robottelo.config import settings
+
+
+def pytest_addoption(parser):
+    """Adds option for enabling external logging"""
+    help_text = '''
+        Flag for enabling promtail on the spawned hosts.
+        This is used for sending of the logs to the external Loki instance
+
+        Usage: --external_logging
+    '''
+    parser.addoption("--external-logging", action="store_true", default=False, help=help_text)
+
+
+def pytest_collection_modifyitems(config):
```

Comment:
I think the way of enabling logging through `pytest_collection_modify` is not the correct way because we are not really modifying the collection items. With that said, we could go with multiple ways of enabling logging:
1. Directly through broker config adding an option to enable external logging and then that seeds in every `Host object` that is being used in ContentHost, Satellite and Capsule classed as it is.
2. Add the option through `robottelo conf` and then enable external logging through ContentHost object and then seed that will be used in Satellite and Capsule classes as it is.
3. Add option through any conf file and write a pytest autouse session level fixture in conftest.py that just seeds what you are doing here in `pytest_collection_modifyitems`.

There might be more smarter ways that asking CI and individuals to pass `--external-logging` option to pytest and individuals might forget that always to pass that option. So having that setting in conf file(either broker or robottelo confs) could be a much better option.

Reply:
> I think the way of enabling logging through pytest_collection_modify is not the correct way because we are not really modifying the collection items.

I'd still rather stick to the hook, as this is a 1-shot function and we don't need to cache it or anything (perfectly fits the hook function description).
For the "incorrect hook" argument - this is easily fixable, as all we need to do is to use a different, more suitable hook, that provides `config` object: `pytest_cmdline_main`.

Not including this inside a config was a little by design, as I'd like to have this feature to be `opt-in`. I believe, the vast minority of users will require the tests to log to external resource and I'd like to prevent people from setting it once in conf and forget about it.

This would generate a lot of logs that will only eat up the storage and won't be used.
from CI perspective, there's nothing easier than appending the option to the pipelines we need this in (we'd need to update CI files even if we used configs anyway).